lib/prop/bucket.rb

```diff
@@ -0,0 +1,72 @@
+require 'prop/key'
+require 'prop/options'
+require 'prop/limiter'
+require 'prop/rate_limited'
+
+module LeakyBucket
+  class Bucket
+
+    class << self
+      def leaky(handle, key = nil, options = {})
+        options, cache_key = prepare(handle, key, options)
+        update_bucket(cache_key, options[:interval], options[:threshold])
+        counter = Prop::Limiter.reader.call(cache_key).fetch(:bucket).to_i
```

Comment:
Couldn't `update_bucket` return the new bucket value, or the `{:bucket => x, :last_updated => y}` hash?